### ISSUE TITLE
Bitcode is deprecated when publishing app to appstore MAUI in 7.0.200-preview.22628.1

### ISSUE THREAD

**Issue by KSemenenko** [dotnet/maui#12863]
### Description

I installed the latest version of VS code for Mac (7.0.200-preview.22628.1 with MAUI workload) and found out that this issue is still there.
https://github.com/dotnet/maui/issues/10989

### Steps to Reproduce

Use a framework containing bitcode

### Link to public reproduction project repository

no

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

IOS

### Did you find any workaround?

```xml
 <PropertyGroup>
    <!-- Properties used to strip bitcode from frameworks when using Visual Studio for Mac -->
    <_StripBitcodeFromFrameworksMasterAfterTargets Condition="'$(OS)'=='Unix'">_ExpandNativeReferences</_StripBitcodeFromFrameworksMasterAfterTargets>
    <_StripBitcodeFromFrameworksMasterDependsOnTargets Condition="'$(OS)'=='Unix'">_StripBitcodeFromFrameworksOnMac</_StripBitcodeFromFrameworksMasterDependsOnTargets>

    <!-- Properties used to strip bitcode from frameworks when using Visual Studio for Windows -->
    <_StripBitcodeFromFrameworksMasterAfterTargets Condition="'$(OS)'!='Unix'">_ComputeFrameworkFilesToPublish</_StripBitcodeFromFrameworksMasterAfterTargets>
    <_StripBitcodeFromFrameworksMasterDependsOnTargets Condition="'$(OS)'!='Unix'">_StripBitcodeFromFrameworksOnWindows</_StripBitcodeFromFrameworksMasterDependsOnTargets>
  </PropertyGroup>
  
  <Target Name="_StripBitcodeFromFrameworksMasterTarget" 
          Condition="'$(_StripBitcodeFromFrameworksMasterDependsOnTargets)'!='' AND @(_FrameworkNativeReference->Count()) != 0" 
          AfterTargets="$(_StripBitcodeFromFrameworksMasterAfterTargets)" 
          DependsOnTargets="$(_StripBitcodeFromFrameworksMasterDependsOnTargets)" />

  <Target Name="_StripBitcodeFromFrameworksOnMac"
          Condition="('$(OutputType)'!='Library' OR '$(IsAppExtension)'=='True') AND '$(_SdkIsSimulator)'=='False'">
    
    <!-- Find the bitcode_strip command -->
    <Exec Command="xcrun -find bitcode_strip" ConsoleToMSBuild="true">
      <Output TaskParameter="ConsoleOutput" PropertyName="_BitcodeStripCommand" />
    </Exec>

    <!-- Strip the bitcode from frameworks -->
    <Exec Command="$(_BitcodeStripCommand) %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)" />
  </Target>

  <Target Name="_StripBitcodeFromFrameworksOnWindows"
          Condition="('$(OutputType)'!='Library' OR '$(IsAppExtension)'=='True') AND '$(_SdkIsSimulator)'=='False' AND '$(IsMacEnabled)'=='true'">

    <!-- Strip the bitcode from frameworks -->
    <Exec SessionId="$(BuildSessionId)"
          Command="&quot;%24(xcrun -find bitcode_strip)&quot; %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)" />

    <CopyFileFromBuildServer 
      SessionId="$(BuildSessionId)" 
      File="%(_FrameworkNativeReference.Identity)" 
      TargetFile="%(_FrameworkNativeReference.Identity)" />    
  </Target>
```
This code helped me. the solution works.
Maybe it should be added by default?

**Comment by PureWeen** [comment]
@SotoiGhost thoughts?

**Comment by SotoiGhost** [comment]
Hi @KSemenenko! 👋🏼 Thank you for your time in reporting this issue.

I think this is a question for the iOS team. The workaround seems harmless, but not sure if this could cause any harm later as this is directly modifying the framework saved on the disk.

@dalexsoto, @rolfbjarne, thoughts on this?

P.S: Regardless of whether this is added, it is best to report the affected framework to its owner so they can fix this from the root.

**Comment by KSemenenko** [comment]
I'm currently doing native binding for agora, they have xcframework.
This is their official sdk, and it seems to work for swift or objective c.

I have this issue when I send the binary to TestFlight.

Also, the experience with bindings in general is not very smooth. It turned out that sln with the bindings library can't be next to maui project. It should be a separate project.

I'm still working on it, but if you need any feedback, I'll be happy to help =)

**Comment by SotoiGhost** [comment]
@KSemenenko you can verify if a framework has bitcode or has the bitcode marker flag enabled by running the following command:

```sh
# Replace the path
otool -l path/to/your/Foo.xcframework/ios-arm64/Foo.framework/Foo | grep __LLVM -1
```

The command should not throw anything if there's no bitcode, otherwise will throw you something like this:

```txt
  cmdsize 152
  segname __LLVM
   vmaddr 0x000000000006c000
--
  sectname __bundle
   segname __LLVM
      addr 0x000000000006c000
```

Hope this helps you!

**Comment by KSemenenko** [comment]
Thanks for your advice, I think now that my sdk is swift.
https://learn.microsoft.com/en-us/xamarin/ios/platform/binding-swift/walkthrough

Will try this one

**Comment by KSemenenko** [comment]
one more questions, https://swiftpackageindex.com/AgoraIO/AgoraRtcEngine_iOS
so this is swift, but if it xcframework I have to anyway create one more swift librarty or I can just use sdk, because thay have h files

**Comment by rolfbjarne** [comment]
> Also, the experience with bindings in general is not very smooth. It turned out that sln with the bindings library can't be next to maui project. It should be a separate project.

Yes, we know it's not very smooth, we have known issues we're working on in this area.

> I think this is a question for the iOS team. The workaround seems harmless, but not sure if this could cause any harm later as this is directly modifying the framework saved on the disk.

Yes, the workaround _seems_ harmless, but I'm hesitant to add code on our side for this, because:

* The app developer can work around it themselves.
* The real fix is for the vendor to fix their SDK to not contain bitcode (since Apple completely deprecated bitcode, at the very least it's useless to ship it).
* It may _seem_ harmless, but there's always a risk.

**Comment by KSemenenko** [comment]
@rolfbjarne thanks for answers!

Well, in fact, for me it is the documentation that is the most difficult. I do not understand the clear steps to be taken. I'm C# developer and have no idea aboout swift and xcode.

So far I have realized that I have to make another library which will be a wrapper for the sdk (and this makes a lot of sense)
But on the other side of the SDK is the same exact library .framework
And according to the documentation, I then have to add to the iOS project a link to the original sdk and a link to my new library-binding-wrapper. But then why do I need an wrapper?

But I'm optimistic and determined to do it =)

**Comment by Syed-RI** [comment]
I think this should be a build task enabled via a flag

**Comment by vplife1** [comment]
I have tried above code lines but show the following error 

The command "/Users/apple/Downloads/Xcode 2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp -r -o /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp" exited with code 127. (MSB3073) 


Please provide me a appropriate solution

**Comment by vplife1** [comment]
I have used the above code snippet but it shows an error follows how to resolve it 
I have tried above code lines but show the following error 

The command "/Users/apple/Downloads/Xcode 2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp -r -o /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp" exited with code 127. (MSB3073) 


Please provide me a appropriate solution

**Comment by rolfbjarne** [comment]
@vplife1 please attach a binlog (https://aka.ms/binlog) that shows the error.

**Comment by vplife1** [comment]
@rolfbjarne   and @Eilon  and @SotoiGhost  I am facing issue when try to add my app on app store it throws the following iisues
We identified one or more issues with a recent delivery for your app, "Calmanac App" 1.0 (3). Please correct the following issues, then upload again.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FBLPromises.framework/FBLPromises' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseCore.framework/FirebaseCore' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseCoreDiagnostics.framework/FirebaseCoreDiagnostics' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseInstallations.framework/FirebaseInstallations' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseMessaging.framework/FirebaseMessaging' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GTMSessionFetcher.framework/GTMSessionFetcher' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleAPIClientForREST.framework/GoogleAPIClientForREST' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleDataTransport.framework/GoogleDataTransport' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleToolboxForMac.framework/GoogleToolboxForMac' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleUtilities.framework/GoogleUtilities' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/Protobuf.framework/Protobuf' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/leveldb.framework/leveldb' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/nanopb.framework/nanopb' contains bitcode.


Please suggest me a correct way how to resolve this issue , I am not able to add my app on app store test flight

**Comment by maonaoda** [comment]
what`s wrong, i can`t build ios device any more.

```
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.8968-net8-rc2\targets\Xamarin.Shared.Sdk.targets(744,3): warning MT7091: 框架 C:/Users/xyz/.nuget/packages/xamarin.firebase.ios.core/8.10.0.3/lib/net6.0-ios15.4/Firebase.Core.resources/GoogleUtilitiesComponents.xcframework/ios-arm64/GoogleUtilitiesComponents.framework 是静态库的框架，不会复制到应用。
1>"$(xcrun -find bitcode_strip)"  -r -o
1>C:\Users\xyz\.nuget\packages\xamarin.firebase.ios.core\8.10.0.3\buildTransitive\Xamarin.Firebase.iOS.Core.targets(216,5): error : /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip: missing argument(s) to: -o option
1>Usage: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip input [-r | -m | -l] [-keep_cs] -o output
1>C:\Users\xyz\.nuget\packages\xamarin.firebase.ios.core\8.10.0.3\buildTransitive\Xamarin.Firebase.iOS.Core.targets(216,5): error MSB3073: 命令“"$(xcrun -find bitcode_strip)"  -r -o ”exit，code is 1。
```

**Comment by vplife1** [comment]
@rolfbjarne please check binlogs
[binlogs.zip](https://github.com/dotnet/maui/files/13069174/binlogs.zip)

Still, I am facing the same issue reply to me ASAP. It's very needful

**Comment by vplife1** [comment]
@maonaoda  Are you facing the same issue 
We identified one or more issues with a recent delivery for your app, "Calmanac App" 1.0 (3). Please correct the following issues, then upload again.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FBLPromises.framework/FBLPromises' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseCore.framework/FirebaseCore' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseCoreDiagnostics.framework/FirebaseCoreDiagnostics' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseInstallations.framework/FirebaseInstallations' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/FirebaseMessaging.framework/FirebaseMessaging' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GTMSessionFetcher.framework/GTMSessionFetcher' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleAPIClientForREST.framework/GoogleAPIClientForREST' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleDataTransport.framework/GoogleDataTransport' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleToolboxForMac.framework/GoogleToolboxForMac' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/GoogleUtilities.framework/GoogleUtilities' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/Protobuf.framework/Protobuf' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/leveldb.framework/leveldb' contains bitcode.

ITMS-90482: Invalid Executable - The executable 'CalmanacMobileAPP.app/Frameworks/nanopb.framework/nanopb' contains bitcode.

**Comment by rolfbjarne** [comment]
@maonaoda 

> what`s wrong, i can`t build ios device any more.
> 
> ```
> 1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.8968-net8-rc2\targets\Xamarin.Shared.Sdk.targets(744,3): warning MT7091: 框架 C:/Users/xyz/.nuget/packages/xamarin.firebase.ios.core/8.10.0.3/lib/net6.0-ios15.4/Firebase.Core.resources/GoogleUtilitiesComponents.xcframework/ios-arm64/GoogleUtilitiesComponents.framework 是静态库的框架，不会复制到应用。
> 1>"$(xcrun -find bitcode_strip)"  -r -o
> 1>C:\Users\xyz\.nuget\packages\xamarin.firebase.ios.core\8.10.0.3\buildTransitive\Xamarin.Firebase.iOS.Core.targets(216,5): error : /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip: missing argument(s) to: -o option
> 1>Usage: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip input [-r | -m | -l] [-keep_cs] -o output
> 1>C:\Users\xyz\.nuget\packages\xamarin.firebase.ios.core\8.10.0.3\buildTransitive\Xamarin.Firebase.iOS.Core.targets(216,5): error MSB3073: 命令“"$(xcrun -find bitcode_strip)"  -r -o ”exit，code is 1。
> ```

Please get a binary build log (binlog) that shows the error - this explains how to get the binlog: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs

**Comment by rolfbjarne** [comment]
@SotoiGhost 

> @rolfbjarne do you think the `xcrun -find bitcode_strip` command is missing some extra quotes due to possible whitespaces?
> 
> ```
> <Exec Command="xcrun -find bitcode_strip" ConsoleToMSBuild="true">
> ```

Not sure

@vplife1 can you get a [binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) that shows the error? That would make it much easier to figure out what's going on on your machine.

**Comment by vplife1** [comment]
Yes, @rolfbjarne this is the issue shown in binlog

/Users/apple/Downloads/Vivek/Projects/CalmanacMobileAPP/CalmanacMobileAPP/CalmanacMobileAPP/CalmanacMobileAPP.csproj(377,5): error MSB3073: The command "/Users/apple/Downloads/Xcode 2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp -r -o /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp" exited with code 127.
    249 Warning(s)
    1 Error(s)

**Comment by vplife1** [comment]
> @SotoiGhost
> 
> > @rolfbjarne do you think the `xcrun -find bitcode_strip` command is missing some extra quotes due to possible whitespaces?
> > ```
> > <Exec Command="xcrun -find bitcode_strip" ConsoleToMSBuild="true">
> > ```
> 
> Not sure
> 
> @vplife1 can you get a [binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) that shows the error? That would make it much easier to figure out what's going on on your machine.

Yes, @rolfbjarne  this is the issue shown in binlog

/Users/apple/Downloads/Vivek/Projects/CalmanacMobileAPP/CalmanacMobileAPP/CalmanacMobileAPP/CalmanacMobileAPP.csproj(377,5): error MSB3073: The command "/Users/apple/Downloads/Xcode 2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp -r -o /Users/apple/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp" exited with code 127.
    249 Warning(s)
    1 Error(s)

**Comment by rolfbjarne** [comment]
I need the entire binlog file (which is a binary format, it's not a text file).

**Comment by maonaoda** [comment]
@rolfbjarne 

**Build task in Xamarin.Firebase.iOS.Core.targets :**
```
  <Target Name="_FirebaseStripBitcodeFromFrameworksOnWindows"
          Condition="'$(IsMacEnabled)'=='true'">
    <!-- Get the frameworks to strip bitcode -->
    <FindInList 
      CaseSensitive="false"
      List="@(_FrameworkNativeReference)"
      ItemSpecToFind="%(_FrameworkNamesToStripBitcode.Identity)"
      MatchFileNameOnly="True">
      <Output TaskParameter="ItemFound" ItemName="_FrameworksToStripBitcode"/>
    </FindInList>

    <!-- Strip the bitcode from frameworks -->
    <Exec SessionId="$(BuildSessionId)"
          Command="&quot;%24(xcrun -find bitcode_strip)&quot; %(_FrameworksToStripBitcode.Identity) -r -o %(_FrameworksToStripBitcode.Identity)" />

    <CopyFileFromBuildServer 
      SessionId="$(BuildSessionId)" 
      File="%(_FrameworksToStripBitcode.Identity)" 
      TargetFile="%(_FrameworksToStripBitcode.Identity)" />    
  </Target>
```
------
the build output of Xamarin.Firebase.iOS.Core.targets was ↓,
and i found that ItemSpecToFind was `FirebaseAnalytics` instead of FirebaseCore, is this right?
because of this, this build task can not get _FrameworksToStripBitcode.Identity,that 
execute  `"$(xcrun -find bitcode_strip)"  -r -o` without any option.

```
1>  任务“FindInList”
1>    Task Parameter:MatchFileNameOnly=True
1>    Task Parameter:ItemSpecToFind=FirebaseAnalytics
1>    Task Parameter:CaseSensitive=False
1>    任务参数:
1>        List=
1>            ...
1>  已完成执行任务“FindInList”的操作。
1>  正在使用程序集“C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.8968-net8-rc2\tools\msbuild\iOS\..\iOS\Xamarin.iOS.Tasks.dll”中的“Exec” Task 。
1>  Task “Exec”
1>    Task Parameter:SessionId=b5afd1a0d7449049c5ecae6315e5c3e9df54befcfae3842e9e1330f2ef5b463b
1>    Task Parameter:Command="$(xcrun -find bitcode_strip)"  -r -o
1>    Exec: 2023-10-24T11:39:09.8837966+09:00 - Started
1>    Exec: 2023-10-24T11:39:09.8837966+09:00 - Initializing
1>    [xma][info]: Trying to get a Build Connection...
1>    [xma][info]: Sending Request Xamarin.Messaging.GetActiveAgentsMessage to topic xma/agents
1>    [xma][info]: Received Response of Xamarin.Messaging.GetActiveAgentsMessage to topic build1763622704wangy/+/xma/agents
1>    [xma][info]: Sending Request Xamarin.Messaging.GetActiveAgentsMessage to topic xma/agents
1>    [xma][info]: Received Response of Xamarin.Messaging.GetActiveAgentsMessage to topic build1763622704wangy/+/xma/agents
1>    [xma][info]: The Build Connection 'Xamarin.Messaging.Build.Client.BuildConnection.17636' is active and ready to use
1>    [xma][info]: Sending Request Xamarin.Messaging.GetActiveAgentsMessage to topic xma/agents
1>    [xma][info]: Received Response of Xamarin.Messaging.GetActiveAgentsMessage to topic build1763622704wangy/+/xma/agents
1>    Exec: 2023-10-24T11:39:09.8997974+09:00 - Initialized
1>    Exec: 2023-10-24T11:39:09.8997974+09:00 - There's no available inputs to copy to the Mac
1>    Exec: 2023-10-24T11:39:09.8997974+09:00 - Serializing inputs
1>    Exec: 2023-10-24T11:39:09.9057976+09:00 - Executing
1>    [xma][info]: Starting remote task execution for 'Mti.Carada.Kenshin.App.Maui': Microsoft.Build.Tasks.Exec
1>    [xma][info]: Sending Request Xamarin.Messaging.Build.Contracts.ExecuteTaskMessage to topic xvs/build/16.4.8968-net8-rc2/execute-task/Mti.Carada.Kenshin.App.Maui/b5afd1a002fExec
1>    [xma][info]: Received Response of Xamarin.Messaging.Build.Contracts.ExecuteTaskMessage to topic build1763622704wangy/+/xvs/build/16.4.8968-net8-rc2/execute-task/Mti.Carada.Kenshin.App.Maui/b5afd1a002fExec
1>    Exec: 2023-10-24T11:39:10.2013326+09:00 - Logging messages
1>    "$(xcrun -find bitcode_strip)"  -r -o
1>    C:\Users\xyz\.nuget\packages\xamarin.firebase.ios.core\8.10.0.3\buildTransitive\Xamarin.Firebase.iOS.Core.targets(216,5): error : /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip: missing argument(s) to: -o option
1>    Usage: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip input [-r | -m | -l] [-keep_cs] -o output
1>    C:\Users\xyz\.nuget\packages\xamarin.firebase.ios.core\8.10.0.3\buildTransitive\Xamarin.Firebase.iOS.Core.targets(216,5): error MSB3073: 命令“"$(xcrun -find bitcode_strip)"  -r -o ”已退出，代码为 1。
1>    [xma][info]: Trying to get a Build Connection...
1>    [xma][info]: Sending Request Xamarin.Messaging.GetActiveAgentsMessage to topic xma/agents
1>    [xma][info]: Received Response of Xamarin.Messaging.GetActiveAgentsMessage to topic build1763622704wangy/+/xma/agents
1>    [xma][info]: Sending Request Xamarin.Messaging.GetActiveAgentsMessage to topic xma/agents
1>    [xma][info]: Received Response of Xamarin.Messaging.GetActiveAgentsMessage to topic build1763622704wangy/+/xma/agents
1>    [xma][info]: The Build Connection 'Xamarin.Messaging.Build.Client.BuildConnection.17636' is active and ready to use
1>    Exec: 2023-10-24T11:39:10.2125162+09:00 - Task execution failed. Disconnecting from the Remote Server...
1>    [xma][info]: Starting disconnection...
1>    [xma][info]: Closing the underlying Messaging connection...
1>    [xma][info]: Closing the underlying Messaging connection...
1>    [xma][info]: Closing the underlying MQTT connection...
1>    [xma][info]: The underlying MQTT connection has been closed.
1>    [xma][info]: Disposing Messaging connection references...
1>    [xma][info]: The underlying Messaging connection has been closed
1>    [xma][info]: Disposing SSH Messaging connection references...
1>    [xma][info]: Starting SSH disconnection...
1>    [xma][info]: Stopped local forwarded port to 54533
1>    [xma][info]: Disconnected SSH client
1>    [xma][info]: Disconnected SCP client
1>    [xma][info]: The underlying Messaging connection has been closed
1>    [xma][info]: Disposing Messaging service references...
1>    Exec: 2023-10-24T11:39:10.2195091+09:00 - Finished
1>  已完成执行任务“Exec”的操作 - 失败。
1>已完成在项目“Mti.Carada.Kenshin.App.Maui.csproj”中生成目标“_FirebaseStripBitcodeFromFrameworksOnWindows”的操作 - 失败。
```

**Comment by rolfbjarne** [comment]
@maonaoda please attach a binlog (https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) file, which contains much more information that's crucial in trying to diagnose this.

**Comment by maonaoda** [comment]
@rolfbjarne 
[CentralNode_devenv_PID=2120_x64_BuildManager_Default.zip](https://github.com/dotnet/maui/files/13111361/CentralNode_devenv_PID.2120_x64_BuildManager_Default.zip)

**Comment by rolfbjarne** [comment]
@maonaoda 

> @rolfbjarne [CentralNode_devenv_PID=2120_x64_BuildManager_Default.zip](https://github.com/dotnet/maui/files/13111361/CentralNode_devenv_PID.2120_x64_BuildManager_Default.zip)

I think this might be an issue with MAX_PATH, where some of the frameworks can't be extracted/created on Windows with the default path locations.

Can you try creating a small root directory (say `C:\x`), and then setting the `XamarinBuildDownloadDir` property to that value in your csproj:

```xml
<PropertyGroup>
    <XamarinBuildDownloadDir>C:\</XamarinBuildDownloadDir>
</PropertyGroup>
```

If that doesn't work, please get an updated binlog.

**Comment by maonaoda** [comment]
@rolfbjarne 
thanks for diagnosing.
but it was not working.
by the way , i had enabled long file support.
and it could deploy into iOS device last week.
But now as long as Xamarin.Firebase.iOS.Core is used, it cannot be deployed。

[2023-10-24-002.zip](https://github.com/dotnet/maui/files/13111933/2023-10-24-002.zip)

**Comment by rolfbjarne** [comment]
@maonaoda does the following directory exist on your Windows machine?

> C:/MSBuildReproLogs/GAppM-8.9.1/GoogleAppMeasurement-8.9.1/Frameworks

**Comment by maonaoda** [comment]
@rolfbjarne 
Yes,
![image](https://github.com/dotnet/maui/assets/32831595/f5e5c04b-08c5-4903-91a3-16f7c4580d5d)

**Comment by rolfbjarne** [comment]
I wonder if you're running into this problem: https://github.com/xamarin/xamarin-macios/issues/19229

Can you try the workaround mentioned here to see if that fixes it: https://github.com/xamarin/xamarin-macios/issues/19229#issuecomment-1777426910?

**Comment by maonaoda** [comment]
@rolfbjarne 
workaround of https://github.com/xamarin/xamarin-macios/issues/19229#issuecomment-1772363165 fixes my problem https://github.com/dotnet/maui/issues/12863#issuecomment-1774854253 
about `xcrun -find bitcode_strip`
thanks you for your work.

**Comment by rolfbjarne** [comment]
Thanks for confirming!

**Comment by maonaoda** [comment]
@rolfbjarne 
I think this happened when i upgraded VS 2022 version 17.8.0 from Preview 3.0 to Preview 4.0 last week

**Comment by vplife1** [comment]
@rolfbjarne , @Eilon @KSemenenko  , @SotoiGhost 
Please Find an attached bin log file 

[CalmanacMobileAPP_Debug_Build_2023-10-25T06_42_14.1545090Z.msbuild.log](https://github.com/dotnet/maui/files/13162289/CalmanacMobileAPP_Debug_Build_2023-10-25T06_42_14.1545090Z.msbuild.log)

@rolfbjarne

**Comment by rolfbjarne** [comment]
@vplife1 you're still using a path to an Xcode with a space in the path: `/Users/apple/Downloads/Xcode 2.app`. 

Can you try and see if this works? I've added quotes to try to support spaces in the Xcode location:

```xml
  <PropertyGroup>
    <!-- Properties used to strip bitcode from frameworks when using Visual Studio for Mac -->
    <_StripBitcodeFromFrameworksMasterAfterTargets Condition="'$(OS)'=='Unix'">_ExpandNativeReferences</_StripBitcodeFromFrameworksMasterAfterTargets>
    <_StripBitcodeFromFrameworksMasterDependsOnTargets Condition="'$(OS)'=='Unix'">_StripBitcodeFromFrameworksOnMac</_StripBitcodeFromFrameworksMasterDependsOnTargets>

    <!-- Properties used to strip bitcode from frameworks when using Visual Studio for Windows -->
    <_StripBitcodeFromFrameworksMasterAfterTargets Condition="'$(OS)'!='Unix'">_ComputeFrameworkFilesToPublish</_StripBitcodeFromFrameworksMasterAfterTargets>
    <_StripBitcodeFromFrameworksMasterDependsOnTargets Condition="'$(OS)'!='Unix'">_StripBitcodeFromFrameworksOnWindows</_StripBitcodeFromFrameworksMasterDependsOnTargets>
  </PropertyGroup>
  
  <Target Name="_StripBitcodeFromFrameworksMasterTarget" 
          Condition="'$(_StripBitcodeFromFrameworksMasterDependsOnTargets)'!='' AND @(_FrameworkNativeReference->Count()) != 0" 
          AfterTargets="$(_StripBitcodeFromFrameworksMasterAfterTargets)" 
          DependsOnTargets="$(_StripBitcodeFromFrameworksMasterDependsOnTargets)" />

  <Target Name="_StripBitcodeFromFrameworksOnMac"
          Condition="('$(OutputType)'!='Library' OR '$(IsAppExtension)'=='True') AND '$(_SdkIsSimulator)'=='False'">
    
    <!-- Find the bitcode_strip command -->
    <Exec Command="xcrun -find bitcode_strip" ConsoleToMSBuild="true">
      <Output TaskParameter="ConsoleOutput" PropertyName="_BitcodeStripCommand" />
    </Exec>

    <!-- Strip the bitcode from frameworks -->
    <Exec Command="&quot;$(_BitcodeStripCommand)&quot; %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)" />
  </Target>

  <Target Name="_StripBitcodeFromFrameworksOnWindows"
          Condition="('$(OutputType)'!='Library' OR '$(IsAppExtension)'=='True') AND '$(_SdkIsSimulator)'=='False' AND '$(IsMacEnabled)'=='true'">

    <!-- Strip the bitcode from frameworks -->
    <Exec SessionId="$(BuildSessionId)"
          Command="&quot;%24(xcrun -find bitcode_strip)&quot; %(_FrameworkNativeReference.Identity) -r -o %(_FrameworkNativeReference.Identity)" />

    <CopyFileFromBuildServer 
      SessionId="$(BuildSessionId)" 
      File="%(_FrameworkNativeReference.Identity)" 
      TargetFile="%(_FrameworkNativeReference.Identity)" />    
  </Target>
```

**Comment by vplife1** [comment]
@rolfbjarne  Yes it works thank you!

**Comment by jfversluis** [comment]
Is there still anything to be done here or can this one be closed?

**Comment by Kbmobi** [comment]
I personally am still running into this issue, and no work arounds provided have been a solution for me. I used to be able to run the simulator and build to a device, but that all stopped once I added in code to interact with firebase for cloud messaging. 

I am running on .Net8, and targeting Xcode 15 with Visual Studio on PC linked to my mac. 

I would like to note that I can still get GitHub Actions to make a release build via MacOs13 runner with Xcode 15.0.1, but I need to be able to debug as well without having to strip out all my push notification/cloud messaging code.
